lib/network/modules/Clustering.js

```diff
@@ -696,6 +693,71 @@ class ClusterEngine {
     return stack;
   }
 
+  /**
+  * Using a clustered nodeId, update with the new options
+  * @param nodeId
+  * @returns {Array}
+  */
+  updateClusteredNode(nodeId, newOptions) {
+    var node = this.body.nodes[nodeId];
+    if (node !== undefined) {
+      node.setOptions(newOptions);
+      this.body.emitter.emit('_dataChanged');
+    }
+  }
+
+  /**
+  * Using a base edgeId, update all related clustered edges with the new options
+  * @param nodeId
+  * @returns {Array}
+  */
+  updateEdge(startEdgeId, newOptions) {
+    let allEdgeIds = this.getClusteredEdges(startEdgeId);
+    for (let i = 0; i < allEdgeIds.length; i++) {
+      var edge = this.body.edges[allEdgeIds[i]];
+      edge.setOptions(newOptions);      
+    }
+    this.body.emitter.emit('_dataChanged');
+  }
+
+  /**
+  * Get a stack of clusterEdgeId's (+base edgeid) that a base edge is the same as. cluster edge C -> cluster edge B -> cluster edge A -> base edge(edgeId)
+  * @param edgeId
+  * @returns {Array}
+  */
+  getClusteredEdges(edgeId) {
+    let stack = [];
+    let max = 100;
```

Comment:
I was using the pre-existing findNode method as a template on how this kind of function should be written, (which contains the 'max' method). I agree an arbitrary value isn't ideal.

You may find this slightly counterintuitive (as i did) but `this.body.edges` is not an array but an object. At the time I also wondered why DataSet was not used to hold the internal representation of the data.